schema-registry/common/src/test/java/com/hortonworks/registries/schemaregistry/retry/RetryExecutorTest.java

```diff
@@ -0,0 +1,131 @@
+import com.hortonworks.registries.schemaregistry.retry.policy.BackoffPolicy;
+import com.hortonworks.registries.util.CustomParameterizedRunner;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+@RunWith(CustomParameterizedRunner.class)
+public class RetryExecutorTest {
+
+    enum RetryPolicyType {
+        FIXED,
+        EXPONENTIAL_BACKOFF;
+    }
+
+    private static RetryPolicyType retryPolicyType;
+
+    @CustomParameterizedRunner.Parameters
+    public static Iterable<RetryPolicyType> profiles() {
+        return Arrays.asList(RetryPolicyType.FIXED, RetryPolicyType.EXPONENTIAL_BACKOFF);
+    }
+
+    @CustomParameterizedRunner.BeforeParam
+    public static void beforeParam(RetryPolicyType retryPolicyTypeFromParameter) throws Exception {
+        retryPolicyType = retryPolicyTypeFromParameter;
+    }
+
+    public RetryExecutorTest(RetryPolicyType retryPolicyType) {
+    }
+
+    @Test
+    public void testForMaxIteration() {
+        AtomicInteger iteration = new AtomicInteger(0);
+
+        createRetryExecutor(100, 3, 100000).execute((() -> {
+            iteration.incrementAndGet();
+            if (iteration.get() < 2) {
+                throw new RuntimeException();
+            } else {
+                return null;
+            }
+        }));
+
+        Assert.assertEquals(2, iteration.get());
+    }
+
+    @Test
+    public void testForMaxTime() {
+        AtomicLong startTime = new AtomicLong(System.currentTimeMillis());
+        AtomicLong stopTime = new AtomicLong(0);
+
+        AtomicInteger iteration = new AtomicInteger(0);
+
+        createRetryExecutor(100, 10000, 2000).execute(() -> {
+            stopTime.set(System.currentTimeMillis());
+            if ((stopTime.get() - startTime.get()) < 1000) {
+                iteration.incrementAndGet();
+                throw new RuntimeException();
+            } else {
+                return null;
+            }
+        });
+
+        Assert.assertTrue((stopTime.get() - startTime.get() < 2000) && iteration.get() > 1);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testExceptionMaxIteration() {
+        createRetryExecutor(100, 1, 60_000).execute(() -> {
+            throw new RuntimeException();
+        });
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testExceptionForMaxSleep() {
+        createRetryExecutor(100, 1000, 300).execute(() -> {
+            throw new RuntimeException();
+        });
+    }
+
+    private RetryExecutor createRetryExecutor(long sleepMs, int iteration, long maxSleepMs) {
+        BackoffPolicy backoffPolicy;
+        Map<String, Object> props = new HashMap<>();
+        switch (retryPolicyType) {
+            case FIXED:
+                props.put(BackoffPolicy.TIMEOUT_MS, maxSleepMs);
```

Comment:
nit: `props.put` can be placed outside of switch statement.

Reply:
Done